tests/integration/cocoon/test_replace_current_transaction_alias.py

```diff
@@ -0,0 +1,134 @@
+            transaction_configuration_data_request=models.TransactionConfigurationDataRequest(
+                aliases=[cls.alias], movements=cls.movements
+            )
+        )
+
+    def test_update_current_alias_with_new_movements(self):
+
+        updated_movements = [
+            models.TransactionConfigurationMovementDataRequest(
+                movement_types="StockMovement",
+                side="Side1",
+                direction=-1,
+                properties={},
+                mappings=[],
+            ),
+            models.TransactionConfigurationMovementDataRequest(
+                movement_types="CashCommitment",
+                side="Side2",
+                direction=1,
+                properties={},
+                mappings=[],
+            ),
+        ]
+
+        replace_current_transaction_alias(
+            api_factory=self.api_factory,
+            updated_alias=self.alias,
+            movement=updated_movements,
+        )
+
+        get_transaction_types = (
+            self.system_configuration_api.list_configuration_transaction_types()
+        )
+
+        uploaded_alias = []
+
+        for trans_type in get_transaction_types.transaction_configs:
+            for alias in trans_type.aliases:
+                if alias.type == new_transaction_type:
+                    uploaded_alias.append(trans_type)
+
+        alias_for_test = models.TransactionConfigurationData(
+            aliases=[self.alias], movements=updated_movements, properties={}
+        )
+        self.assertEqual(uploaded_alias[0].to_dict(), alias_for_test.to_dict())
+
+    def test_update_alias_which_does_not_exist(self):
+
+        updated_movements = [
+            models.TransactionConfigurationMovementDataRequest(
+                movement_types="StockMovement",
+                side="Side1",
+                direction=-1,
+                properties={},
+                mappings=[],
+            ),
+            models.TransactionConfigurationMovementDataRequest(
+                movement_types="CashCommitment",
+                side="Side2",
+                direction=1,
+                properties={},
+                mappings=[],
+            ),
+        ]
+
+        updated_alias = models.TransactionConfigurationTypeAlias(
+            type=create_scope_id().replace("-", ""),
+            description="TESTBUY1",
+            transaction_class="TESTBUY1",
+            transaction_group="SYSTEM1",
+            transaction_roles="AllRoles",
+        )
+
+        with self.assertRaises(Warning) as error:
+            replace_current_transaction_alias(
+                api_factory=self.api_factory,
+                updated_alias=updated_alias,
+                movement=updated_movements,
+            )
+
+        self.assertEqual(
+            error.exception.args[0], "The requested alias is not available in LUSID."
+        )
```

Comment:
What is the behaviour here? Does it still successfully update the transaction type configuration even if the alias does not exist?

Reply:
The intended behavior - the function request will fail if you try and amend a transaction type which does not exist.
Here is my thinking:
- Transaction types are system wide settings 
- Incorrectly amending them can have bad consequences
- Therefore I would like users to be very explicit when using the function 
- This function is specifically for updating a current transaction type configuration
-  If that transaction type does not exist, I would rather we tell the user (via this error) rather than create the transaction type